File: composeApp/src/commonTest/kotlin/com/hereliesaz/hg2gui/commands/main/raw/CalcTest.java
package com.hereliesaz.hg2gui.commands.main.raw;

import org.junit.Test;
import static org.junit.Assert.*;

import com.hereliesaz.hg2gui.commands.ExecutePack;

public class CalcTest {

    @Test
    public void testSimpleAddition() throws Exception {
        calc cmd = new calc();
        ExecutePack pack = new ExecutePack(null) {};
        pack.args = new Object[]{"2+2"};

        String result = cmd.exec(pack);
        assertEquals("4.0", result);
    }

    @Test
    public void testMultiplication() throws Exception {
        calc cmd = new calc();
        ExecutePack pack = new ExecutePack(null) {};
        pack.args = new Object[]{"3*5"};

        String result = cmd.exec(pack);
        assertEquals("15.0", result);
    }

    @Test
    public void testSqrt() throws Exception {
        calc cmd = new calc();
        ExecutePack pack = new ExecutePack(null) {};
        pack.args = new Object[]{"sqrt(9)"};

        String result = cmd.exec(pack);
        assertEquals("3.0", result);
    }

    @Test
    public void testComplex() throws Exception {
        calc cmd = new calc();
        ExecutePack pack = new ExecutePack(null) {};
        pack.args = new Object[]{"2+3*4"};

        String result = cmd.exec(pack);
        assertEquals("14.0", result);
    }
}

File: composeApp/src/commonTest/kotlin/com/hereliesaz/hg2gui/commands/main/raw/CalcTest.kt
package com.hereliesaz.hg2gui.commands.main.raw

import com.hereliesaz.hg2gui.PlatformContext
import com.hereliesaz.hg2gui.commands.ExecutePack
import org.junit.Assert.assertEquals
import org.junit.Test

class CalcTest {

    private val mockPack = object : ExecutePack() {
        override val context: PlatformContext = object : PlatformContext {
            override fun getString(resId: Int): String = ""
            override fun getString(resId: Int, vararg args: Any): String = ""
        }
        override fun getPrefsSave(): Any? = null
        override fun getLaunchInfo(): Any? = null
    }

    @Test
    fun testSimpleAddition() {
        val cmd = calc()
        mockPack.args = arrayOf("2+2")
        mockPack.currentIndex = 0

        val result = cmd.exec(mockPack)
        assertEquals("4.0", result)
    }

    @Test
    fun testMultiplication() {
        val cmd = calc()
        mockPack.args = arrayOf("3*5")
        mockPack.currentIndex = 0

        val result = cmd.exec(mockPack)
        assertEquals("15.0", result)
    }

    @Test
    fun testSqrt() {
        val cmd = calc()
        mockPack.args = arrayOf("sqrt(9)")
        mockPack.currentIndex = 0

        val result = cmd.exec(mockPack)
        assertEquals("3.0", result)
    }

    @Test
    fun testComplex() {
        val cmd = calc()
        mockPack.args = arrayOf("2+3*4")
        mockPack.currentIndex = 0

        val result = cmd.exec(mockPack)
        assertEquals("14.0", result)
    }
}
